source/simulator/melt.cc

```diff
@@ -0,0 +1,1090 @@
+                       (parameters.use_locally_conservative_discretization)
+                       ?
+                       std_cxx11::shared_ptr<FiniteElement<dim> >(new FE_DGP<dim>(parameters.stokes_velocity_degree-1))
+                       :
+                       std_cxx11::shared_ptr<FiniteElement<dim> >(new FE_Q<dim>(parameters.stokes_velocity_degree-1)),
+                       1,
+                       1));
+
+    variables.insert(variables.begin()+3,
+                     VariableDeclaration<dim>("fluid velocity",
+                                              std_cxx11::shared_ptr<FiniteElement<dim> >(
+                                                new FE_Q<dim>(parameters.stokes_velocity_degree)),
+                                              dim,
+                                              1));
+
+  }
+
+
+  template <int dim>
+  void
+  MeltHandler<dim>::
+  declare_parameters (ParameterHandler &prm)
+  {
+    prm.enter_subsection ("Melt settings");
+
+    prm.declare_entry ("Melt transport threshold", "1e-3",
+                       Patterns::Double (),
+                       "The porosity limit for melt migration. For smaller porosities, the equations "
+                       "reduce to the Stokes equations and do neglect melt transport. Only used "
+                       "if Include melt transport is true. ");
+    prm.declare_entry ("Heat advection by melt", "false",
+                       Patterns::Bool (),
+                       "Whether to use a porosity weighted average of  the melt and solid velocity "
+                       "to advect heat in the temperature equation or not. If this is set to true, "
+                       "additional terms are assembled on the left-hand side of the temperature "
+                       "advection equation. Only used if Include melt transport is true."
+                       "If this is set to false, only the solid velocity is used (as in models "
+                       "without melt migration).");
+    prm.leave_subsection();
+
+    FluidPressureBoundaryConditions::declare_parameters<dim> (prm);
+  }
+
+  template <int dim>
+  MeltHandler<dim>::MeltHandler (ParameterHandler &prm)
+  {
+    parse_parameters(prm);
+  }
+
+  template <int dim>
+  void
+  MeltHandler<dim>::initialize_simulator (const Simulator<dim> &simulator_object)
+  {
+    this->SimulatorAccess<dim>::initialize_simulator(simulator_object);
+    if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(fluid_pressure_boundary_conditions.get()))
+      sim->initialize_simulator (simulator_object);
+    fluid_pressure_boundary_conditions->initialize ();
+  }
+
+  template <int dim>
+  void
+  MeltHandler<dim>::
+  parse_parameters (ParameterHandler &prm)
+  {
+    prm.enter_subsection ("Melt settings");
+    this->melt_transport_threshold = prm.get_double("Melt transport threshold");
+    this->heat_advection_by_melt = prm.get_bool("Heat advection by melt");
```

Comment:
But you could add braces around the subsection block as in other `parse_parameters` functions.